### PR TITLE
java_keystore: overwrite instead of fail when password or alias does not match

### LINKED ISSUE #1671: java_keystore: crashes when keystore password is changed
##### SUMMARY
See the disabled test added in #1666. If it is enabled, it will crash the module.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
java_keystore

##### ANSIBLE VERSION
```paste below
all
```

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #1671. Completes what @quidame started in #2183.

##### ISSUE TYPE
- Bugfix Pull Request
- Feature Pull Request

##### COMPONENT NAME
java_keystore

### CODE REVIEW

File: plugins/modules/system/java_keystore.py [hunk 1]
```diff
@@ -60,7 +60,9 @@
     description:
       - Password that should be used to secure the keystore.
       - If the provided password fails to unlock the keystore, the module
-        fails. This behavior will change in a next release.
+        will re-create the keystore with the new passphrase. This behavior
+        changed in community.general 3.0.0, before the module would fail
```
Comment (aminvakil) [inline plugins/modules/system/java_keystore.py:64]:
```suggestion
        changed in community.general 3.0.0, before that the module would fail
```

**Comment by felixfontein** [comment]
I missed in the documentation / changelog fragment that this also changes the behavior w.r.t. alias mismatch. I've now updated it.

**Comment by felixfontein** [comment]
@quidame I would be happy if you could give this a review :)

File: tests/unit/plugins/modules/system/test_java_keystore.py [hunk 2]
```diff
@@ -256,13 +256,31 @@ def test_cert_changed_fail_alias_does_not_exist(self):
...
+            name='foo',
+            password='changeit'
+        ))
+
+        module = AnsibleModule(
+            argument_spec=self.spec.argument_spec,
+            supports_check_mode=self.spec.supports_check_mode
+        )
+
+        module.fail_json = Mock()
```
Comment (quidame) [inline tests/unit/plugins/modules/system/test_java_keystore.py:276]: `module.fail_json()` should not be reached. If it is mocked, what happens if it is reached ? Maybe this one line alone makes no difference, but I think it is useless.
```suggestion
```
The same in `test_cert_changed_fail_alias_does_not_exist()` above.

File: tests/unit/plugins/modules/system/test_java_keystore.py [hunk 3]
```diff
@@ -256,13 +256,30 @@ def test_cert_changed_fail_alias_does_not_exist(self):
...
+            name='foo',
+            password='changeit'
+        ))
+
+        module = AnsibleModule(
+            argument_spec=self.spec.argument_spec,
+            supports_check_mode=self.spec.supports_check_mode
+        )
+
+
```
Comment (felixfontein) [inline tests/unit/plugins/modules/system/test_java_keystore.py:276]:
```suggestion
```

**Review by quidame — APPROVED** [review]
@felixfontein thanks for fixing this ;)

**Comment by felixfontein** [comment]
@aminvakil @quidame thanks a lot for reviewing!
@quidame also thanks for preparing this one so it was so easy to fix ;-)

**Pull request merged**